src/toxcore/dht/server.rs

```diff
@@ -1411,6 +1451,130 @@ mod tests {
         assert!(alice.handle_packet((packet, addr)).wait().is_err());
     }
 
+    // handle_announce_request
+    #[test]
+    fn server_handle_announce_request_test() {
+        let (alice, precomp, bob_pk, _bob_sk, rx, addr) = create_node();
+
+        let sendback_data = 42;
+        let payload = AnnounceRequestPayload {
+            ping_id: initial_ping_id(),
+            search_pk: gen_keypair().0,
+            data_pk: gen_keypair().0,
+            sendback_data
+        };
+        let inner = InnerAnnounceRequest::new(&precomp, &bob_pk, payload);
+        let onion_return = OnionReturn {
+            nonce: gen_nonce(),
+            payload: vec![42; ONION_RETURN_3_PAYLOAD_SIZE]
+        };
+        let packet = DhtPacket::AnnounceRequest(AnnounceRequest {
+            inner,
+            onion_return: onion_return.clone()
+        });
+
+        assert!(alice.handle_packet((packet, addr)).wait().is_ok());
+
+        let (received, _rx) = rx.into_future().wait().unwrap();
+        let (packet, addr_to_send) = received.unwrap();
+
+        assert_eq!(addr_to_send, addr);
+
+        let response = unpack!(packet, DhtPacket::OnionResponse3);
+
+        assert_eq!(response.onion_return, onion_return);
+
+        let response = unpack!(response.payload, InnerOnionResponse::AnnounceResponse);
+
+        assert_eq!(response.sendback_data, sendback_data);
+
+        let payload = response.get_payload(&precomp).unwrap();
+
+        assert_eq!(payload.announce_status, AnnounceStatus::Failed);
```

Comment:
Are there other tests that handle the other variants of `AnnounceStatus` enum?

Reply:
There are, inside onion_announce module.